<h1>Analysis on App Store and Google Play</h1>


### Objective
Our aim of this project is to identify the different attributes that make an app profitable on both App Store and Google Play.
This project is to aid and assist app developers and marketing teams in making a data driven decisions with respect to the apps attended to be built.
We assume that the teams are interested in building a free app where the main source of revenue is in-app ads. Hence, our analysis will be primarily on free apps and to understand which apps has high interaction rate and attract more users.

### Exploring the Data

As of September 2018, there were approximately 2 million iOS apps available on the App Store, and 2.1 million Android apps on Google Play.

Due to limited time and resources required to collect all data consist of four million apps is not available, we decided to collect and find representing data instead. Lucklly, we have found a free cost data sample which is suitable for our purpose. 

- [IOS Apps](https://www.kaggle.com/ramamet4/app-store-apple-data-set-10k-apps) data set consisting of approximately seven thousand apps. You can download the data set directly from [this link](https://www.kaggle.com/ramamet4/app-store-apple-data-set-10k-apps).
- [Android Apps](https://www.kaggle.com/lava18/google-play-store-apps/data#) data set consisting of approximately ten thousand apps. you can download the data set directly from [this link](https://www.kaggle.com/lava18/google-play-store-apps/data#).

Now we start our exploratory analysis by loading the data sets.

In [1]:
from csv import reader

## ios data set
openfile = open('playstore.csv', encoding="utf8")
readfile = reader(openfile)
android = list(readfile)
android_header = android[0] # data headers as list
android = android[1:] # rows as list of lists

## android store data set
openfile = open('appstore.csv', encoding="utf8") 
readfile = reader(openfile)
ios = list(readfile)
ios_header = ios[0] # data headers as list
ios = ios[1:] # rows as list of lists
print('Data sets are loaded!')

Data sets are loaded!


######  **A function to explore the data**
Here we make a function to explore the data and navigate through data sets whenever is required. Also, we will add an option to our function that show the number of rows and columns. 

In [2]:
# fuction to displaly rows and columns when called
def explore_data(dataset, start, end, rows_columns=False): 
    dataset_slice = dataset[start:end]
    for row in dataset_slice:
        print(row)
        print('\n')
        
    if rows_columns:
        print('number of rows--->', len(dataset))
        print('numbers of columns---->', len(dataset[0]))

In [3]:
print(android_header)
print ('\n')
explore_data(android, 0, 3, True)

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['Coloring book moana', 'ART_AND_DESIGN', '3.9', '967', '14M', '500,000+', 'Free', '0', 'Everyone', 'Art & Design;Pretend Play', 'January 15, 2018', '2.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


number of rows---> 10841
numbers of columns----> 13


From android apps data set:
- We got headers of the data set and a sample of entries.
- Our data sample has **10841** entries. 
- Our data sample has **13** attributes.

In [4]:
print(ios_header)
print('\n')
explore_data(ios, 0, 3, True)

['', 'id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


['1', '281656475', 'PAC-MAN Premium', '100788224', 'USD', '3.99', '21292', '26', '4', '4.5', '6.3.5', '4+', 'Games', '38', '5', '10', '1']


['2', '281796108', 'Evernote - stay organized', '158578688', 'USD', '0', '161065', '26', '4', '3.5', '8.2.2', '4+', 'Productivity', '37', '5', '23', '1']


['3', '281940292', 'WeatherBug - Local Weather, Radar, Maps, Alerts', '100524032', 'USD', '0', '188583', '2822', '3.5', '4.5', '5.0.0', '4+', 'Weather', '37', '5', '3', '1']


number of rows---> 7197
numbers of columns----> 17


From ios apps data set:
- We got headers of the data set and a sample of entries.
- Our data sample has **7197** entries. 
- Our data sample has **17** attributes.

### Cleaning the data

Once we have loaded the data, we need to confirm data integrity for further processing. Possible issues we might find with the data might be (missing values, duplicate entries, wrong values, wrong data types,...etc). A great start will be reading through [the discussion section](https://www.kaggle.com/lava18/google-play-store-apps/discussion/66015) of the data source where we found an issue with the data in Play Store. 

In there, the community has found that there is an entry missing in attribute 'Rating' where in entry number 10472 has a rating of 19 which is not possible because the maximum rating for an app in play store is 5. This issue is caused by a missing value. To fix that, we have decided to remove that entry. 

#### 1- Removing a row with a wrong value

In [5]:
print(len(android))
del android[10472] # removing row 10472 
print(len(android))

10841
10840


#### 2- Removing duplicate entries
In this step, we will start exploring data sets more to indentify if any duplicate entries are present. 

In [6]:
# counting duplicate entries
import operator
total = 0
orignal = {}
for i in android: # accessing apps in android list
    name = i[0]
    if name in orignal: # generation of frequency table
        total=total+1
        orignal[name]+=1
    else:
        orignal[name]=1

sigma = 0        
duplicate = {}
for x,y in orignal.items(): # checking for duplicates then save in duplicate variable
    if y>1:
        duplicate[x] = y
        sigma+=1
        
top_duplicate_app = max(duplicate.items(), key=operator.itemgetter(1))[0]
num_duplicates = sigma
length = len(android)
percent = str((num_duplicates/length)*100)

print(str(total) + ' number of double entries in data set is')
print(str(length)+ ' number of apps in the data set')
print(str(num_duplicates) + ' number of duplicats in the data set')
print(percent + '% of apps are duplicates in data set')
print('top duplicate app is: '+ top_duplicate_app + ' with ' + str(duplicate[top_duplicate_app]) + ' entries')

1181 number of double entries in data set is
10840 number of apps in the data set
798 number of duplicats in the data set
7.361623616236162% of apps are duplicates in data set
top duplicate app is: ROBLOX with 9 entries


From above, we have found there is **798** unique apps with duplicate entries. 
In total, **1181** duplicate entries which need to be removed as we don't want them to be included in our analysis.
One way is to start removing apps based on an attribute where we minimize loss and increase our gain from the data.

In [7]:
# displaying duplicate entries for instagram as example
print(android_header)
print('\n')
for app in android:
    name = app[0]
    if name =='Instagram':
        print(app)

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66577446', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66509917', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']


As we can see using intagram as an example of an app with duplicate entry.
We find that it is using the attribute **'Reviews'** is most sutible choice 
for removing duplicates. Depending on the number of reviews an app received, the app with highest reviews will be kept.
After removing duplicates, expected number of entries left in the data set is **10840 - 1181 = (9659) entries**.

In [8]:
# using review attribute to map duplicates and calculate rows left
review = {} # will contain duplicate apps with highest reviews
for app in android: # accessing rows in list
    name = app[0]
    n_reviews = float(app[3])
    if n_reviews in review and n_review>review[name]:
        review[name] = n_reviews
    elif n_reviews not in review:
        review[name] = n_reviews
print('Actual length of data set :---->' , len(review)) # rows left from mapping
print('Expected length: ------>' , len(android)-total)  # rows left calulated from pervious test duplicates=1181

Actual length of data set :----> 9659
Expected length: ------> 9659


Next, we will use the filter above contain duplicate apps only with the highest reviews
to filter out android data set and have a clean set we can work within our analyse.

In [9]:
# filltering only apps with highest review and add to a list of lists
android_clean = [] # contain only apps with highest reviews
added = [] 
for app in android:
    name = app[0]
    n_review = float(app[3])
    if n_review == review[name] and name not in added: 
        android_clean.append(app)
        added.append(name)
print(len(android_clean))

9659


#### 3- Removing Non-English Apps

Through exploring both data sets, we find that some of the apps don't intend for an English language audience.

In [10]:
# example of non engligsh apps
arr = []
for app in android_clean:
    name = app[0]
    for char in name:
        if ord(char) >127: 
            arr.append(app)
print(len(arr))
print(arr[50:55])

1045
[['Flame - درب عقلك يوميا', 'EDUCATION', '4.6', '56065', '37M', '1,000,000+', 'Free', '0', 'Everyone', 'Education', 'July 26, 2018', '3.3', '4.1 and up'], ['Flame - درب عقلك يوميا', 'EDUCATION', '4.6', '56065', '37M', '1,000,000+', 'Free', '0', 'Everyone', 'Education', 'July 26, 2018', '3.3', '4.1 and up'], ['Kanji test · Han search Kanji training (free version)', 'EDUCATION', '4.2', '6736', '22M', '1,000,000+', 'Free', '0', 'Mature 17+', 'Education', 'October 17, 2017', '2.51.0', '4.0.3 and up'], ['GRE® Flashcards by Kaplan', 'EDUCATION', '4.0', '316', '29M', '50,000+', 'Free', '0', 'Everyone', 'Education', 'February 15, 2017', '1.4', '4.0 and up'], ['Babbel – Learn Spanish', 'EDUCATION', '4.4', '54798', '11M', '1,000,000+', 'Free', '0', 'Everyone', 'Education', 'July 30, 2018', '20.7.2', '4.4 and up']]


As we can see, some apps are not English and some have special characters and a mix of languages. Though we have found **1045** non-English apps in the play store with above lines, it is not the perfect solution, but it would be good enough to continue with further.
Next, we will use the lines above to filter out both data sets.

In [11]:
# fuction to map non_english apps takes app name as an argument
def if_english(string):
    non_ascii = 0
    
    for character in string:
        if ord(character) > 127:
            non_ascii += 1
    
    if non_ascii > 3:
        return False
    else:
        return True

In [12]:
# using the function to fillter english android apps and add them to a list
android_english = []
for app in android_clean:
    name = app[0]
    if if_english(name):
        android_english.append(app)

In [13]:
# using the function to fillter english ios apps and add them to a list 
ios_english = []
for app in ios:
    name = app[2]
    if if_english(name):
        ios_english.append(app)

In [14]:
explore_data(android_english, 0,3, True)
print('\n\n\n')
explore_data(ios_english, 0,3, True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


number of rows---> 9614
numbers of columns----> 13




['1', '281656475', 'PAC-MAN Premium', '100788224', 'USD', '3.99', '21292', '26', '4', '4.5', '6.3.5', '4+', 'Games', '38', '5', '10', '1']


['2', '281796108', 'Evernote - stay organized', '158578688', 'USD', '0', '161065', '26', '4', '3.5', '8.2.2', '4+', 'Productivity', '37', '5', '23', '1']


['3', '281940292', 'WeatherBug - Local Weather, Radar, Maps, Alerts', '100524032', 'USD',

#### 4- Selecting free apps only
As mentioned in the introduction, our engineering and marketing team are interested in only building 
a free app. Hence, we will analyze free apps from both data sets. We need to isolate those free apps 
from the rest.

In [15]:
print(android_header)
print(ios_header)

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']
['', 'id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


In [16]:
# finding free apps and add them to lists
android_set = []
ios_set = []

for app in android_english:
    prices = app[7]
    if prices == '0':
        android_set.append(app)
for app in ios_english:
    prices = app[5]
    if prices == '0':
        ios_set.append(app)
        
print(len(android_set))
print(len(ios_set))

8864
3222


For android apps , we are left with only **8864** entries in the data set.

For ios apps however, we are left with only **3222** entries in the data set.

## Analyzing the Data

After we finished with loading and cleaning the data to forms that we can work with, in this part we start asking questions related to our goal. Our goal is to derive an understanding about our data and possibly find patterns will aid us to draw conclusions. We would like to know based on our data, which apps are most popular in both sets?, based on (reviews, ratings, number of downloads,... etc.).

Also, we are interested in finding categories and genre of those popular apps. And based on those findings. we can focus on categories and genre which have better opportunities for the application the team would like to build.

For better understanding and clarity, a function to generate frequency tables and percentages and another function to display tables are needed. Below, we will use these two functions thoroughly.

In [17]:
# a function returns apps distibution in specified attribute or a column
def freq_table(dataset, index): # takes data set and attribute as arguments
    table = {}
    total = 0
    
    for row in dataset:
        value = row[index]
        total+=1
        if value in table:
            table[value]+=1
        else:
            table[value]=1
            
    table_percent = {}
    for key in table:
        percentage = (table[key]/ total)*100
        table_percent[key] = percentage
    return table_percent # percent of apps in each unique value in atrribute

def display_table(dataset,index): # function will print value of attribute and percent of total apps 
    table = freq_table(dataset,index)
    table_dis = []
    for key in table:
        store_val_tuple = (table[key], key)
        table_dis.append(store_val_tuple)
        
    sort_table = sorted(table_dis, reverse= True)
    for val in sort_table:
        print(val[1], ' : ', val[0])

In [18]:
display_table(android_set,1) # percent of total apps in  'Catogory' attribute

FAMILY  :  19.223826714801444
GAME  :  9.510379061371841
TOOLS  :  8.461191335740072
BUSINESS  :  4.580324909747293
LIFESTYLE  :  3.9034296028880866
PRODUCTIVITY  :  3.892148014440433
FINANCE  :  3.7003610108303246
MEDICAL  :  3.5424187725631766
SPORTS  :  3.4183212996389893
PERSONALIZATION  :  3.3167870036101084
COMMUNICATION  :  3.2490974729241873
HEALTH_AND_FITNESS  :  3.068592057761733
PHOTOGRAPHY  :  2.944494584837545
NEWS_AND_MAGAZINES  :  2.7978339350180503
SOCIAL  :  2.6624548736462095
TRAVEL_AND_LOCAL  :  2.33528880866426
SHOPPING  :  2.2450361010830324
BOOKS_AND_REFERENCE  :  2.1435018050541514
DATING  :  1.861462093862816
VIDEO_PLAYERS  :  1.782490974729242
MAPS_AND_NAVIGATION  :  1.3989169675090252
FOOD_AND_DRINK  :  1.2409747292418771
EDUCATION  :  1.128158844765343
LIBRARIES_AND_DEMO  :  0.9363718411552346
AUTO_AND_VEHICLES  :  0.9250902527075812
ENTERTAINMENT  :  0.8799638989169676
HOUSE_AND_HOME  :  0.8235559566787004
WEATHER  :  0.8009927797833934
EVENTS  :  0.71074007

In [19]:
display_table(android_set,-4) # percent of total apps in  'genre' attribute

Tools  :  8.449909747292418
Entertainment  :  6.069494584837545
Education  :  5.347472924187725
Business  :  4.580324909747293
Productivity  :  3.892148014440433
Lifestyle  :  3.892148014440433
Finance  :  3.7003610108303246
Medical  :  3.5424187725631766
Sports  :  3.463447653429603
Personalization  :  3.3167870036101084
Communication  :  3.2490974729241873
Action  :  3.1024368231046933
Health & Fitness  :  3.068592057761733
Photography  :  2.944494584837545
News & Magazines  :  2.7978339350180503
Social  :  2.6624548736462095
Travel & Local  :  2.3240072202166067
Shopping  :  2.2450361010830324
Books & Reference  :  2.1435018050541514
Simulation  :  2.0419675090252705
Dating  :  1.861462093862816
Arcade  :  1.861462093862816
Video Players & Editors  :  1.782490974729242
Casual  :  1.7486462093862816
Maps & Navigation  :  1.3989169675090252
Food & Drink  :  1.2409747292418771
Puzzle  :  1.128158844765343
Racing  :  0.9927797833935018
Role Playing  :  0.9363718411552346
Libraries & Dem

In [20]:
display_table(ios_set,-5) # percent of total apps in  'Genres' attribute

Games  :  58.16263190564867
Entertainment  :  7.883302296710118
Photo & Video  :  4.9658597144630665
Education  :  3.662321539416512
Social Networking  :  3.2898820608317814
Shopping  :  2.60707635009311
Utilities  :  2.5139664804469275
Sports  :  2.1415270018621975
Music  :  2.0484171322160147
Health & Fitness  :  2.0173805090006205
Productivity  :  1.7380509000620732
Lifestyle  :  1.5828677839851024
News  :  1.3345747982619491
Travel  :  1.2414649286157666
Finance  :  1.1173184357541899
Weather  :  0.8690254500310366
Food & Drink  :  0.8069522036002483
Reference  :  0.5586592178770949
Business  :  0.5276225946617008
Book  :  0.4345127250155183
Navigation  :  0.186219739292365
Medical  :  0.186219739292365
Catalogs  :  0.12414649286157665


### 1-  Most common categories and genres for each market in precentages :

**Android apps by category:**
- Family = **19.22%** 
- Games = **9.51%** 
- Tools = **8.46%** 

**Note**

Though, the difference between category and genre in Play Store is not clear, by comparison we have found that category is more general capturing the big picture that we need. 

**Ios apps by genre:**
- Games = **58.16%** 
- Entertament = **7.88%** 
- Photos & videos = **4.96%** 
- Education = **3.66%**
- Social & Networking = **3.28%**

In Play Store market, it's clear that productivity related apps (LifeStyle, Finance, Medical, Tools, sports, Communication) made up around **41%** of all other genres. For Family and Games however, made up around **30%**. We understand that Family and Games are dominating the market place. If we search for free apps on Family category, games apps for children will show up.

App Store was a different story. We found that Games alone accounted for **58%** of all free apps in another genre. That wasn't the case in the Play Store where the apps were more spread across different genre.

### 2- Top apps in most common genres and categories:

In [21]:
print('App Store attributes','------>',ios_header)
print('\n')
print('Play Store attributes','------>', android_header)

App Store attributes ------> ['', 'id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


Play Store attributes ------> ['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


#### A- App Store apps


In [22]:
# calculating the average of ratings each unique genres recieved using nested loop
rating_per_genre = freq_table(ios_set, -5)
sort_list = []
for genre in rating_per_genre: # for loop to access unique genres
    total = 0
    len_genre = 0
    
    for app in ios_set: # for loop to extract actuall number of ratings for each genre
        name = app[2]
        ios_genre = app[-5]
        if ios_genre == genre:
            num_rating = float(app[6])
            total+=num_rating
            len_genre+=1         
    avg_rating = (total/len_genre)
    sort_list.append((avg_rating,genre))
    top_ratings = sorted(sort_list, reverse=True)
    
for genres in top_ratings: # printing the result i
    print(genres[1],' : ', genres[0])

Navigation  :  86090.33333333333
Reference  :  74942.11111111111
Social Networking  :  71548.34905660378
Music  :  57326.530303030304
Weather  :  52279.892857142855
Book  :  39758.5
Food & Drink  :  33333.92307692308
Finance  :  31467.944444444445
Photo & Video  :  28441.54375
Travel  :  28243.8
Shopping  :  26919.690476190477
Health & Fitness  :  23298.015384615384
Sports  :  23008.898550724636
Games  :  22788.6696905016
News  :  21248.023255813954
Productivity  :  21028.410714285714
Utilities  :  18684.456790123455
Lifestyle  :  16485.764705882353
Entertainment  :  14029.830708661417
Business  :  7491.117647058823
Education  :  7003.983050847458
Catalogs  :  4004.0
Medical  :  612.0


#### Average app ratings by genre in  App Store:
- Navigation **86,090**
- Referance **74,942**
- Social Networking **71,942**
- Music **57,326**
- Weather **52,279**
- Book **39,758**

In [23]:
# function to display number of apps in each genre and total ratings
# the function will help explore multipule genres 
def app_rating(data_set, genre):
    listed = []
    total = 0
    for app in ios_set:
        if app[-5] == genre:
            listed.append((int(app[6]), app[2]))
            total+=1
    print('number of apps in ' + genre ,'---------> ', total)
    print('\n')
    
    sort_listed = sorted(listed, reverse=True)
    for apps in sort_listed:
        print(apps[1], ':' , apps[0])
    
app_rating(ios_set, 'Navigation' )

number of apps in Navigation --------->  6


Waze - GPS Navigation, Maps & Real-time Traffic : 345046
Google Maps - Navigation & Transit : 154911
Geocaching® : 12811
CoPilot GPS – Car Navigation & Offline Maps : 3582
ImmobilienScout24: Real Estate Search in Germany : 187
Railway Route Search : 5


In [24]:
app_rating(ios_set, 'Reference' )

number of apps in Reference --------->  18


Bible : 985920
Dictionary.com Dictionary & Thesaurus : 200047
Dictionary.com Dictionary & Thesaurus for iPad : 54175
Google Translate : 26786
Muslim Pro: Ramadan 2017 Prayer Times, Azan, Quran : 18418
New Furniture Mods - Pocket Wiki & Game Tools for Minecraft PC Edition : 17588
Merriam-Webster Dictionary : 16849
Night Sky : 12122
City Maps for Minecraft PE - The Best Maps for Minecraft Pocket Edition (MCPE) : 8535
LUCKY BLOCK MOD ™ for Minecraft PC Edition - The Best Pocket Wiki & Mods Installer Tools : 4693
GUNS MODS for Minecraft PC Edition - Mods Tools : 1497
Guides for Pokémon GO - Pokemon GO News and Cheats : 826
WWDC : 762
Horror Maps for Minecraft PE - Download The Scariest Maps for Minecraft Pocket Edition (MCPE) Free : 718
VPN Express : 14
Real Bike Traffic Rider Virtual Reality Glasses : 8
教えて!goo : 0
Jishokun-Japanese English Dictionary & Translator : 0


In [25]:
app_rating(ios_set, 'Social Networking')

number of apps in Social Networking --------->  106


Facebook : 2974676
Pinterest : 1061624
Skype for iPhone : 373519
Messenger : 351466
Tumblr : 334293
WhatsApp Messenger : 287589
Kik : 260965
ooVoo – Free Video Call, Text and Voice : 177501
TextNow - Unlimited Text + Calls : 164963
Viber Messenger – Text & Call : 164249
Followers - Social Analytics For Instagram : 112778
MeetMe - Chat and Meet New People : 97072
We Heart It - Fashion, wallpapers, quotes, tattoos : 90414
InsTrack for Instagram - Analytics Plus More : 85535
Tango - Free Video Call, Voice and Chat : 75412
LinkedIn : 71856
Match™ - #1 Dating App. : 60659
Skype for iPad : 60163
POF - Best Dating App for Conversations : 52642
Timehop : 49510
Find My Family, Friends & iPhone - Life360 Locator : 43877
Whisper - Share, Express, Meet : 39819
Hangouts : 36404
LINE PLAY - Your Avatar World : 34677
WeChat : 34584
Badoo - Meet New People, Chat, Socialize. : 34428
Followers + for Instagram - Follower Analytics : 28633
GroupMe : 28

In [26]:
app_rating(ios_set, 'Music')

number of apps in Music --------->  66


Pandora - Music & Radio : 1126879
Spotify Music : 878563
Shazam - Discover music, artists, videos & lyrics : 402925
iHeartRadio – Free Music & Radio Stations : 293228
SoundCloud - Music & Audio : 135744
Magic Piano by Smule : 131695
Smule Sing! : 119316
TuneIn Radio - MLB NBA Audiobooks Podcasts Music : 110420
Amazon Music : 106235
SoundHound Song Search & Music Player : 82602
Sonos Controller : 48905
Bandsintown Concerts : 30845
Karaoke - Sing Karaoke, Unlimited Songs! : 28606
My Mixtapez Music : 26286
Sing Karaoke Songs Unlimited with StarMaker : 26227
Ringtones for iPhone & Ringtone Maker : 25403
Musi - Unlimited Music For YouTube : 25193
AutoRap by Smule : 18202
Spinrilla - Mixtapes For Free : 15053
Napster - Top Music & Radio : 14268
edjing Mix:DJ turntable to remix and scratch music : 13580
Free Music - MP3 Streamer & Playlist Manager Pro : 13443
Free Piano app by Yokee : 13016
Google Play Music : 10118
Certified Mixtapes - Hip Hop Albums 

In [27]:
app_rating(ios_set, 'Book')

number of apps in Book --------->  14


Kindle – Read eBooks, Magazines & Textbooks : 252076
Audible – audio books, original series & podcasts : 105274
Color Therapy Adult Coloring Book for Adults : 84062
OverDrive – Library eBooks and Audiobooks : 65450
HOOKED - Chat Stories : 47829
BookShout: Read eBooks & Track Your Reading Goals : 879
Dr. Seuss Treasury — 50 best kids books : 451
Green Riding Hood : 392
Weirdwood Manor : 197
MangaZERO - comic reader : 9
謎解き2016 : 0
謎解き : 0
ikouhoushi : 0
MangaTiara - love comic reader : 0


In [28]:
app_rating(ios_set, 'Games')

number of apps in Games --------->  1874


Clash of Clans : 2130805
Temple Run : 1724546
Candy Crush Saga : 961794
Angry Birds : 824451
Subway Surfers : 706110
Solitaire : 679055
CSR Racing : 677247
Crossy Road - Endless Arcade Hopper : 669079
Injustice: Gods Among Us : 612532
Hay Day : 567344
PAC-MAN : 508808
DragonVale : 503230
Head Soccer : 481564
Despicable Me: Minion Rush : 464312
The Sims™ FreePlay : 446880
Sonic Dash : 418033
8 Ball Pool™ : 416736
Tiny Tower - Free City Building : 414803
Jetpack Joyride : 405647
Bike Race - Top Motorcycle Racing Games : 405007
Kim Kardashian: Hollywood : 397730
Trivia Crack : 393469
WordBrain : 391401
Sniper 3D Assassin: Shoot to Kill Gun Game : 386521
Flow Free : 373857
Geometry Dash Lite : 370370
▻Sudoku : 359832
Fruit Ninja® : 327025
Pixel Gun 3D : 301182
Temple Run 2 : 295211
My Horse : 293857
Word Cookies! : 287095
Dragon City Mobile : 277268
The Simpsons™: Tapped Out : 274501
Plants vs. Zombies™ 2 : 267394
Clash Royale : 266921
Pokémon GO 

DC Legends : 16460
FarmVille: Tropic Escape - Harvest in Paradise : 16457
Sonic CD : 16358
Plants vs. Zombies™ Heroes : 16305
Farm Heroes Super Saga: Match, Collect and Grow! : 16259
The Walking Dead: Road to Survival - Strategy Game : 16116
Tap Tap Fish - Abyssrium : 15876
Forge of Empires : 15617
Ninja Kid Run VR: Runner & Racing Games For Free : 15399
FIFA Mobile Soccer : 15365
Cupcake Mania™ : 15336
Hovercraft: Takedown - Custom Combat Cars : 15290
Flappy Golf : 14995
Arrow : 14673
Zombie Highway 2 : 14593
Traffic Rider : 14592
Six! : 14505
Fruit Nibblers : 14505
Deer Hunter 2017 : 14461
Bottle Flip 2k16 : 14263
Hovercraft - Build Fly Retry : 14123
Top Eleven 2017 - Be a Soccer Manager : 14104
Snakeio : 13870
WWE SuperCard : 13842
94% : 13805
Transformers: Earth Wars : 13664
Slots - Huuuge Casino: Slot Machines : 13637
Pokémon TCG Online : 13559
my KONAMI Slots - Vegas Casino Slot Machine Games : 13345
KleptoCats : 13300
Churchill Solitaire - World War Card Game : 13241
Farm Story 

Gear.Club - True Racing : 2793
Flow Free: Hexes : 2791
Galaxy Reavers-Space Strategy game(RTS) : 2778
Juju on the Beat : 2772
World of Cubes Survival Craft : 2758
Big Bang Racing : 2758
Pocket Politics : 2748
Amateur Surgeon 4 : 2743
Tennis Champs Returns : 2740
Ingress : 2739
Slots: VIP Deluxe Slots Machines - Free Slot Games : 2732
Mahjong Solitaire Star! Your Favorite Game! : 2731
Talking Tom Bubble Shooter : 2659
Christmas Sweeper 3 : 2645
Super Atomic : 2638
Star Trek ® - Wrath of Gems : 2636
Bakery Blitz: Cooking Game : 2605
Pixel Car Racer : 2602
Can You Dab? : 2601
Bits of Sweets : 2572
Jelly Blast: New Exciting Match 3 : 2557
Soccer Clicker : 2556
Pirate Power : 2555
Glowing Snake King - Anaconda Diep War Battle Game : 2554
Mine Blitz : 2541
Athletics 2: Summer Sports - Free : 2541
Just Dance Now : 2535
Fear the Walking Dead: Dead Run–Tactical Runner : 2533
Torque Burnout : 2531
League of War: Mercenaries : 2523
VOEZ : 2508
Design This Castle : 2500
OPUS: The Day We Found Eart

Golf Island : 702
Hello Kitty Lunchbox – Food Maker : 694
Dictator 2 : 694
Football Heroes PRO 2017 - featuring NFL Players : 691
Catch Rate : 691
Bubble Buggie : 691
Demi Lovato - Zombarazzie Adventure : 690
Escalate : 689
Gas Station Car Parking Simulator a Real Road Racing Park Game : 687
Diamond Story: Jewelry Quest : 683
Lumo Deliveries : 679
Stickman Ice Hockey : 676
caRRage : 671
Super Slime Blitz – Gumball Endless Arcade Climber : 668
Mannequin Challenge : 668
Kubo: A Samurai Quest ™ - Match, Collect, Battle! : 668
Blocky Soccer - Endless Arcade Runner : 667
Smash Slots : 663
Bridge Construction Simulator 3D a Real City Building Physics Sim : 653
Mystic Castle - the Simplest & Best RPG and Adventure Game : 650
BARRIER X : 650
Gravity Trump : 647
Endless Ducker : 639
Psych : 635
LINE PokoPoko : 631
Bike Unchained : 629
TIME LOCKER - Shooter : 627
Girls PJ Party - Dress Up, Spa & Fun : 621
Baby Boss - Care, Dress Up and Play : 616
Super Jabber Jump : 612
Titan Brawl : 611
String 

Bumper Jump : 166
3D Dubai Parking Simulator Drive Real Extreme Super Sports Car : 166
Magic Nightfall : 164
Color Bots : 164
ENYO : 163
Multi Level Car Parking 6 Shopping Mall Garage Lot : 162
Flick Golf Extreme : 161
Real Cake Maker 3D - Bake, Design & Decorate : 160
Jumpy Tree : 160
Tentacles - Enter the Mind : 159
Soda Coin Party: Free Casino Pusher : 159
Road to be King : 158
8-BIT WATERSLIDE : 158
Pixelgrams: Pixel Puzzles : 156
Snake splix : 155
Respeck on my Name : 155
PICTLOGICA FINAL FANTASY : 154
Duck Roll : 154
Championship Manager 17 : 154
Obstacle Course Extreme Car Parking Simulator : 153
Baby Bottle Challenge - Water Bottle Flip : 152
Yummy Birthday - Party Food Maker : 151
Super Smash the Office - Endless Destruction! : 151
Cops - On Patrol : 151
Helmet Virtual Reality 3D Joke : 150
Dash Masters : 149
Mikey Jumps : 148
Sweet Tales : 147
Impulse GP - Super Bike Racing : 147
Fancy Cakes - Endless Tasty Puzzle : 147
Fish Fist : 146
Zap Zombies : 145
Spider Solitaire ∙ : 1

のび毛 : 0
striker – dropping balls : 0
real happy majiang-all stars majiang,mahjong : 0
When you clear one,million yen CHANCE!! : 0
Uncolored Boy : 0
Toy Blocks Game for Brain Traning -TSUMIKI- : 0
Tiny Bone : 0
The Pocket Quiz for Pokemon : 0
Tetro Blocks - Beat IQ with Tetrix-style-puzzle. : 0
Tetminos - Classic Puzzle Game : 0
Tapping Monsters : 0
TLMJ, Tout Le Monde Joue : 0
Super Knock and Run : 0
Sumikko gurashi-Puzzling Ways : 0
Stress Release Grenade : 0
Soccer Puzzle for Brain Training -SoccerStrike- : 0
Simple Reversi Online : 0
Shark Boom - Challenge Friends with your Pet : 0
Samurai Ninja Puzzle ONIMARU : 0
SUDOKU -The puzzle game that makes your brain younger!- : 0
SPEED:3D Racing : 0
S, - Puzzle Game With Create Level Feature : 0
Reversi REAL - Multiplayer Board game : 0
Refreshing ! Brain training fit : 0
REVERSI ZERO - free classic game : 0
Puzzle Game for Brain Training!! PITATTO!! : 0
Puzzle Game for Brain Training -KITINTO- : 0
Poker - Texas Holdem HD Poker : 0
Pokelog

In [29]:
app_rating(ios_set, 'Entertainment')

number of apps in Entertainment --------->  254


Netflix : 308844
Fandango Movies - Times + Tickets : 291787
Colorfy: Coloring Book for Adults : 247809
IMDb Movies & TV - Trailers and Showtimes : 183425
TRUTH or DARE!!! - FREE : 171055
Mad Libs : 117889
Twitch : 109549
Action Movie FX : 101222
Voice Changer Plus : 98777
iFunny :) : 98344
The CW : 97368
The Moron Test : 88613
DIRECTV : 81006
ABC – Watch Live TV & Stream Full Episodes : 78890
Xbox : 72187
Redbox : 60236
Talking Tom Cat 2 for iPad : 56399
Hulu: Watch TV Shows & Stream the Latest Movies : 56170
NBC – Watch Now and Stream Full TV Episodes : 55950
Emoji> : 55338
DIRECTV App for iPad : 47506
Amazon Prime Video : 43667
CBS Full Episodes and Live TV : 39436
FOX NOW - Watch Full Episodes and Stream Live TV : 39391
Talking Angela for iPad : 32763
Recolor - Coloring Book : 31180
Talking Ben the Dog for iPad : 31116
Talking Tom Cat for iPad : 29492
YouTube Kids : 28560
Tom's Love Letters : 27711
HBO GO : 26278
NFL Sunday Ticket : 

**App Store Summery:**

Results from the previous sections revealed that about 51% of free apps in the app store are listed under Game genre. On the other hand, a number number of apps didn't illustrate the full picture. After exploring further, game apps received more ratings than any other genres. (Clash of Clans= **2130805**, Temple Run: **1724546**, Candy Crush Saga= **961794**). The total number of those apps ratings make up about 5 million, which explains the popularity.
Those ratings might not be a good indicator because of the massive amount of game apps and the fact that the ratings are way far from the average **22 thousand** ratings only.
Also, genres were not so popular in a number of apps like (Referance, Book, navigations) had high average ratings than top three genres.

Although apps in Navigation and Music genres aren't in the focus because they don't fit the model of high interaction rate to make a free app profitable, yet Referance and Book genres are a good fit for the app the teams would like to build. Reference and Book genres have less apps listed on both (Referance= **18 apps**, Book= **14 apps**) which makes penetrating the market a lot easier compared with entering for example game genre and also offers less risk.
That is a clear opportunity to explored if features such as game feature or interactive part of social networking like chatting or share reviews and thoughts got added to the app to increase overall interaction as well as benefiting from the popularity of games and social networking.

#### B- Play Store Apps

In [30]:
display_table(android_set,5)

1,000,000+  :  15.749097472924186
100,000+  :  11.563628158844766
10,000,000+  :  10.503158844765343
10,000+  :  10.209837545126353
1,000+  :  8.393501805054152
100+  :  6.915613718411552
5,000,000+  :  6.825361010830325
500,000+  :  5.561823104693141
50,000+  :  4.7721119133574
5,000+  :  4.512635379061372
10+  :  3.5424187725631766
500+  :  3.2490974729241873
50,000,000+  :  2.3014440433213
100,000,000+  :  2.1322202166064983
50+  :  1.917870036101083
5+  :  0.78971119133574
1+  :  0.5076714801444043
500,000,000+  :  0.2707581227436823
1,000,000,000+  :  0.22563176895306858
0+  :  0.04512635379061372
0  :  0.01128158844765343


In [31]:
# calculating number of installs each category recieved
android_category= freq_table(android_set, 1)
listed = []

for category in android_category:
    total = 0
    len_category = 0
    
    for app in android_set: # extracting number of installs 
        set_category = app[1]
        installs = app[5]
        if set_category == category:
            installs = installs.replace(',', '') # removing unwanted characters from number of installs
            installs = installs.replace('+', '')
            total+=float(installs) # convert str type to float to find total
            len_category+=1
    avg_installs = total/len_category
    listed.append((avg_installs, category))
sorted_list = sorted(listed, reverse=True)

for category in sorted_list: # print the resul
    print(category[1], ':', category[0])

COMMUNICATION : 38326063.197916664
VIDEO_PLAYERS : 24790074.17721519
SOCIAL : 23253652.127118643
PHOTOGRAPHY : 17840110.40229885
PRODUCTIVITY : 16772838.591304347
TRAVEL_AND_LOCAL : 13984077.710144928
GAME : 12914435.883748516
TOOLS : 10801391.298666667
NEWS_AND_MAGAZINES : 9549178.467741935
ENTERTAINMENT : 9146923.076923076
BOOKS_AND_REFERENCE : 8767811.894736841
SHOPPING : 7036877.311557789
PERSONALIZATION : 5201482.6122448975
FAMILY : 5180161.789906103
WEATHER : 5074486.197183099
SPORTS : 4274688.722772277
HEALTH_AND_FITNESS : 4167457.3602941176
MAPS_AND_NAVIGATION : 4056941.7741935486
ART_AND_DESIGN : 1986335.0877192982
FOOD_AND_DRINK : 1924897.7363636363
EDUCATION : 1768500.0
BUSINESS : 1704192.3399014778
LIFESTYLE : 1437816.2687861272
FINANCE : 1387692.475609756
HOUSE_AND_HOME : 1331540.5616438356
DATING : 854028.8303030303
COMICS : 817657.2727272727
AUTO_AND_VEHICLES : 647317.8170731707
LIBRARIES_AND_DEMO : 638503.734939759
PARENTING : 542603.6206896552
BEAUTY : 513151.886792452

#### Average app installs per genre in Play Store
- COMMUNICATION  **38,326,063.197**
- VIDEO_PLAYERS  **24,790,074.17**
- SOCIAL **23,253,652.12**
- PHOTOGRAPHY **17,840,110.40**
- PRODUCTIVITY **16,772,838.59**
- GAME **12,914,435.88**
- FAMILY **5,180,161.78**

In [32]:
# fuction to display average number of apps installed in each category 
def android_installs(data_set, genre):
    listed = []
    original = []
    for app in data_set:
        name = app[0]
        category = app[1]
        installs = app[5]
        installs = installs.replace(',','')
        installs = installs.replace('+', '')
        
        if category == genre and (installs == '1000000000' or installs == '500000000' or installs == '100000000'):
            listed.append((int(installs), name))
        if category == genre:
            original.append(name)
            
    sorted_list = sorted(listed, reverse=True)
    percente = (len(sorted_list)/len(original))*100
    
    print('- there are ', len(sorted_list), 'apps in ', genre, 'downloaded over 100 million times', 'accounting for ', percente,'%')
    print('- our set has', len(original), 'apps in',genre )
    print('\n')
    
    for apps in sorted_list:
        print(apps[1], ':', apps[0])

In [33]:
android_installs(android_set, 'COMMUNICATION') 

- there are  27 apps in  COMMUNICATION downloaded over 100 million times accounting for  9.375 %
- our set has 288 apps in COMMUNICATION


WhatsApp Messenger : 1000000000
Skype - free IM & video calls : 1000000000
Messenger – Text and Video Chat for Free : 1000000000
Hangouts : 1000000000
Google Chrome: Fast & Secure : 1000000000
Gmail : 1000000000
imo free video calls and chat : 500000000
Viber Messenger : 500000000
UC Browser - Fast Download Private & Secure : 500000000
LINE: Free Calls & Messages : 500000000
Google Duo - High Quality Video Calls : 500000000
imo beta free calls and text : 100000000
Yahoo Mail – Stay Organized : 100000000
Who : 100000000
WeChat : 100000000
UC Browser Mini -Tiny Fast Private & Secure : 100000000
Truecaller: Caller ID, SMS spam blocking & Dialer : 100000000
Telegram : 100000000
Opera Mini - fast web browser : 100000000
Opera Browser: Fast and Secure : 100000000
Messenger Lite: Free Calls & Messages : 100000000
Kik : 100000000
KakaoTalk: Free Calls & Tex

In [34]:
android_installs(android_set, 'VIDEO_PLAYERS') 

- there are  9 apps in  VIDEO_PLAYERS downloaded over 100 million times accounting for  5.69620253164557 %
- our set has 158 apps in VIDEO_PLAYERS


YouTube : 1000000000
Google Play Movies & TV : 1000000000
MX Player : 500000000
VivaVideo - Video Editor & Photo Movie : 100000000
VideoShow-Video Editor, Video Maker, Beauty Camera : 100000000
VLC for Android : 100000000
Motorola Gallery : 100000000
Motorola FM Radio : 100000000
Dubsmash : 100000000


In [35]:
android_installs(android_set, 'SOCIAL') 

- there are  13 apps in  SOCIAL downloaded over 100 million times accounting for  5.508474576271186 %
- our set has 236 apps in SOCIAL


Instagram : 1000000000
Google+ : 1000000000
Facebook : 1000000000
Snapchat : 500000000
Facebook Lite : 500000000
VK : 100000000
Tumblr : 100000000
Tik Tok - including musical.ly : 100000000
Tango - Live Video Broadcast : 100000000
Pinterest : 100000000
LinkedIn : 100000000
Badoo - Free Chat & Dating App : 100000000
BIGO LIVE - Live Stream : 100000000


In [36]:
android_installs(android_set, 'PHOTOGRAPHY')

- there are  19 apps in  PHOTOGRAPHY downloaded over 100 million times accounting for  7.2796934865900385 %
- our set has 261 apps in PHOTOGRAPHY


Google Photos : 1000000000
Z Camera - Photo Editor, Beauty Selfie, Collage : 100000000
YouCam Perfect - Selfie Photo Editor : 100000000
YouCam Makeup - Magic Selfie Makeovers : 100000000
Sweet Selfie - selfie camera, beauty cam, photo edit : 100000000
S Photo Editor - Collage Maker , Photo Collage : 100000000
Retrica : 100000000
PicsArt Photo Studio: Collage Maker & Pic Editor : 100000000
PhotoGrid: Video & Pic Collage Maker, Photo Editor : 100000000
Photo Editor Pro : 100000000
Photo Editor Collage Maker Pro : 100000000
Photo Collage Editor : 100000000
LINE Camera - Photo editor : 100000000
Cymera Camera- Photo Editor, Filter,Collage,Layout : 100000000
Candy Camera - selfie, beauty camera, photo editor : 100000000
Camera360: Selfie Photo Editor with Funny Sticker : 100000000
BeautyPlus - Easy Photo Editor & Selfie Camera : 100000000
B612 -

In [37]:
android_installs(android_set, 'PRODUCTIVITY') 

- there are  22 apps in  PRODUCTIVITY downloaded over 100 million times accounting for  6.3768115942028984 %
- our set has 345 apps in PRODUCTIVITY


Google Drive : 1000000000
Microsoft Word : 500000000
Google Calendar : 500000000
Dropbox : 500000000
Cloud Print : 500000000
WPS Office - Word, Docs, PDF, Note, Slide & Sheet : 100000000
SwiftKey Keyboard : 100000000
Samsung Notes : 100000000
Microsoft PowerPoint : 100000000
Microsoft Outlook : 100000000
Microsoft OneNote : 100000000
Microsoft OneDrive : 100000000
Microsoft Excel : 100000000
Google Slides : 100000000
Google Sheets : 100000000
Google Keep : 100000000
Google Docs : 100000000
Evernote – Organizer, Planner for Notes & Memos : 100000000
ES File Explorer File Manager : 100000000
ColorNote Notepad Notes : 100000000
CamScanner - Phone PDF Creator : 100000000
Adobe Acrobat Reader : 100000000


In [38]:
android_installs(android_set, 'BOOKS_AND_REFERENCE') 

- there are  5 apps in  BOOKS_AND_REFERENCE downloaded over 100 million times accounting for  2.631578947368421 %
- our set has 190 apps in BOOKS_AND_REFERENCE


Google Play Books : 1000000000
Wattpad 📖 Free Books : 100000000
Bible : 100000000
Audiobooks from Audible : 100000000
Amazon Kindle : 100000000


In [39]:
android_installs(android_set, 'FAMILY')

- there are  26 apps in  FAMILY downloaded over 100 million times accounting for  1.5258215962441315 %
- our set has 1704 apps in FAMILY


Google Play Games : 1000000000
Pou : 500000000
My Talking Tom : 500000000
Candy Crush Saga : 500000000
Where's My Water? Free : 100000000
Tom Loves Angela : 100000000
Talking Tom Cat 2 : 100000000
Talking Tom Cat : 100000000
Talking Tom & Ben News : 100000000
Talking Ginger : 100000000
Talking Ben the Dog : 100000000
ROBLOX : 100000000
Plants vs. Zombies FREE : 100000000
Netflix : 100000000
My Talking Angela : 100000000
Minion Rush: Despicable Me Official Game : 100000000
Hay Day : 100000000
Flow Free : 100000000
Farm Heroes Saga : 100000000
Duolingo: Learn Languages Free : 100000000
Cut the Rope FULL FREE : 100000000
Clash of Clans : 100000000
Clash Royale : 100000000
Bubble Witch 2 Saga : 100000000
Bitmoji – Your Personal Emoji : 100000000
Angry Birds 2 : 100000000


**Google Play summery:**

For communication, the average installs have the highest average installs of **38 million** installs. 6 apps had around a billion installs and other 5 apps had **500 million** installs of total apps downloaded over **500 million**, they make up 21%. Since Google owns Play Store, it was expected they would offer products and compete, they had two products each had a billion installs. The 11 apps which had over 500 million installs, influence the mean to be high which doesn't reflect the real underlying reason for such popularity.


For video_player category, however, we notice that only three apps are in **500 million** to billion installs. Two apps which are google products had over a billion installs which also explain the reason for their high average we obtained earlier. Also, the average doesn't clearly demonstrate the reason for such popularity since only three apps are extremely popular.

Social category shares the same characteristics as video_players category where only handful of extremely popular apps had over 500 million install making only **5.5%** of all apps in this category.

In the case of photography, only one app also owned by google capturing **51%** of installs for apps over 100 million. Although we have 261 apps, yet one externally popular app influenced high installs average.

Productivity is yet another example of apps which are extremely popular dominating over **500 million** installs portion of this category. Google and Microsoft with their 12 apps, had over half of this segment. Obviously they would have a huge influence on the average as well.

Books_and_reference has only 5 apps over **100 million** installs. But, only one app received billion installs and four are above **100 million** benchmark we set. Meaning the rest of 185 apps have a number of installs which are more spread out.

For family category, only four apps received **500 million** and over of installs. Also looking closer, we notice that compared with other categories, more apps in the family category received 100 million downloads. Furthermore, apps which seem to be games in nature are present in this category which we have discussed in earlier section.

## Conclusion
Throughout this project, we analyzed both App Store and Play Store and we found out great amount of information. The objective of this project was to identify what makes an app popular among users in general and how that popularity translated in number of ratings and installs. Apps built by FAANG (Facebook, Apple, Amazon, Google) is a lot more popular, hence receive huge ratings and installs due to resource available to them. As engineering and marketing team focus on building free apps with in-app ad model, we focused away from FAANG companies with the possibility to study different features from their apps in the future. 

Considering the risks associated with building an app, it's a good idea to start with a simple version and gradually add more features along the way after the concept has been proved. Looking back to our analysis, categories such as **Book, REFERANCE, Family and PRODUCTIVITY** has high potential to attract users and has relatively low penetration risk, as there isn't so many apps listed but has enough number of ratings and installs on average for both markets.

From above, making an app for adults or kids that combine features from above category may have high potential for success. Furthermore, research is needed to be carried to identify attributes that increase content engagement in the app. Making an app that offers books that has a dictionary and element of learning vocabulary as well as gaming like environment might seem a good idea as it captures categories identified as popular and also has high engagment. 